test/e2e-go/upgrades/stateproof_test.go

```diff
@@ -0,0 +1,261 @@
+	client := fixture.AlgodClient
+
+	curStatus, err := client.Status()
+	a.NoError(err)
+	curRound := curStatus.LastRound
+
+	// the below window controls the likelihood a block will be proposed by the account under test
+	// since block proposer selection is probabilistic, it is not guaranteed that the account will be chosen
+	// it is a trade-off between test flakiness and test duration
+	for window > 0 {
+		window--
+		curRound++
+		err := fixture.WaitForRoundWithTimeout(curRound)
+		a.NoErrorf(err, "fixture failed waiting for round %d", curRound)
+
+		// See if account was participating by looking at block proposers
+		blockWasProposed := fixture.VerifyBlockProposed(account, 1)
+		if blockWasProposed {
+			return blockWasProposed
+		}
+	}
+	return false
+}
+
+// This test starts with participation keys in Version29, then attempts to let the richest user participate even after
+//  consensus upgrade.
+func TestParticipationWithoutStateProofKeys(t *testing.T) {
+	a := require.New(fixtures.SynchronizedTest(t))
+
+	consensus := getStateProofConcensus()
+
+	var fixture fixtures.RestClientFixture
+	fixture.SetConsensus(consensus)
+	fixture.Setup(t, filepath.Join("nettemplates", "TwoNodesWithoutStateProofPartkeys.json"))
+	defer fixture.Shutdown()
+	// Want someone to start with a key to participate with...
+
+	act, err := fixture.GetRichestAccount()
+	a.NoError(err)
+
+	var address = act.Address
+
+	runUntilProtocolUpgrades(a, &fixture)
+
+	a.NotEmpty(address)
+
+	proposalWindow := 50 // giving 50 rounds to participate, should be able to participate every second round.
+	blockWasProposedByPartkeyOnlyAccountRecently := waitForAccountToProposeBlock(a, &fixture, address, proposalWindow)
+	a.True(blockWasProposedByPartkeyOnlyAccountRecently, "partkey-only account should be proposing blocks")
+}
+
+func TestLargeKeyRegistration(t *testing.T) {
+	a := require.New(fixtures.SynchronizedTest(t))
+
+	consensus := getStateProofConcensus()
+
+	tmp := config.Consensus[protocol.ConsensusFuture]
+	newVer := config.Consensus[protocol.ConsensusFuture]
+	newVer.CompactCertRounds = 64
+	config.Consensus[protocol.ConsensusFuture] = newVer
+	defer func() { config.Consensus[protocol.ConsensusFuture] = tmp }()
+
+	var fixture fixtures.RestClientFixture
+	fixture.SetConsensus(consensus)
+	fixture.Setup(t, filepath.Join("nettemplates", "TwoNodesWithoutStateProofPartkeys.json"))
+	defer fixture.Shutdown()
+
+	runUntilProtocolUpgrades(a, &fixture)
+
+	txid, err := registerKeyInto("Node", &fixture, a, 300000, protocol.ConsensusFuture)
+	a.NoError(err)
+
+	client := fixture.LibGoalClient
+
+	for {
+		// Check if we know about the transaction yet
+		txn, err := client.PendingTransactionInformation(txid)
+		a.NoError(err)
+
+		if txn.ConfirmedRound > 0 {
+			break
+		}
+
+		a.Empty(txn.PoolError)
+	}
+}
+
+func TestCompactCertificatesAreCreatedAfterVersionUpgrade(t *testing.T) {
```

Comment:
move this to compact crt tests. 
+ I can't find a keyreg transaction. how does it work?